test/bin/stop_api_test_server.sh

```diff
@@ -19,9 +19,49 @@ _get_osmt_project_dir() {
   echo "${project_dir}"
 }
 
+
+get_app_pid(){
+  local app_pid; app_pid="$(jps -l | grep ${OSMT_APP_CLASS} | awk '{print $1}')"
+
+  echo "${app_pid}"
+}
+
+check_pid_status_retry(){
+  for retry_limit in {12..0}; do
+    if [[ -z "$(get_app_pid)" ]]; then
+      echo "INFO: Application ${OSMT_APP_CLASS} successfully stopped. Exiting..."
+      exit 0
+    fi
+    echo "INFO: Could not stop application ${OSMT_APP_CLASS}. Will retry ${retry_limit} more times. Retrying in 3 seconds..."
+    local app_pid; app_pid="$(jps -l | grep ${OSMT_APP_CLASS} | awk '{print $1}')"
+    echo "${app_pid}"
+    sleep 3
+  done
+
+  echo "ERROR: Could not stop ${OSMT_APP_CLASS} after 12 retries. Exiting..." 1>&2;
+  exit 135
+}
+
+shutdown_osmt_app(){
+  echo "INFO: Attempting to stop application ${OSMT_APP_CLASS}."
+  local app_pid; app_pid="$(get_app_pid)"
+
+  if [[ -z "${app_pid}" ]]; then
+    echo "ERROR: Application ${OSMT_APP_CLASS} not running" 1>&2;
+    exit 135
+  else
+    echo "INFO: Application ${OSMT_APP_CLASS} found PID ${app_pid}."
+  fi
+
+  kill -9 "${app_pid}"
+
+  check_pid_status_retry
+}
+
 project_dir="$(_get_osmt_project_dir)" || exit 135
 
 # stop the API test server
+shutdown_osmt_app
 
 # clean up API test Docker resources
 
```

Comment:
Probably need to shut down the Docker stack for the API tests. Dev team can decide if you should destroy the ES and MySQL containers or not. Either way, shouldn't destroy your local dev containers.